### ISSUE TITLE
jvm crash when profiling (openjdk 11)

### ISSUE THREAD

**Issue by choxsword** [async-profiler/async-profiler#709]
```
#
# A fatal error has been detected by the Java Runtime Environment:
#
#  Internal Error (codeCache.cpp:631), pid=12861, tid=20947
#  guarantee(result == NULL || !result->is_zombie() || result->is_locked_by_vm() || VMError::is_error_reported()) failed: unsafe access to zombie method


V  [libjvm.so+0xe178e5]  VMError::report_and_die(int, char const*, char const*, __va_list_tag*, Thread*, unsigned char*, void*, void*, char const*, int, unsigned long)+0x1c5
V  [libjvm.so+0xe18807]  VMError::report_and_die(Thread*, void*, char const*, int, char const*, char const*, __va_list_tag*)+0x47
V  [libjvm.so+0x69abea]  report_vm_error(char const*, int, char const*, char const*, ...)+0xfa
V  [libjvm.so+0x5f5e30]  CodeCache::find_blob(void*)+0xb0
V  [libjvm.so+0xdbb650]  JavaThread::pd_get_top_frame_for_signal_handler(frame*, void*, bool)+0x140
V  [libjvm.so+0x757b7a]  AsyncGetCallTrace+0x11a
C  [libasyncProfiler.so+0x1d373]  Profiler::getJavaTraceAsync(void*, ASGCT_CallFrame*, int)+0x8c3
C  [libasyncProfiler.so+0x1d83f]  Profiler::recordSample(void*, unsigned long long, int, Event*)+0x20f
C  [libasyncProfiler.so+0x1ec6d]  PerfEvents::signalHandler(int, siginfo_t*, void*)+0xdd


```

**Comment by apangin** [comment]
Can you show entire `hs_err.log`? (it's ok to obfuscate it if needed)

**Comment by richardstartin** [comment]
@choxsword upgrading to 11.0.17 should resolve your issue, this is caused by a bug in AsyncGetCallTrace (not async-profiler) which was fixed by Datadog in [JDK-8283849](https://bugs.openjdk.org/browse/JDK-8283849).

**Comment by apangin** [comment]
As mentioned above, this is a JVM bug that has already been fixed.
Also, you can workaround the bug in async-profiler by adding `--cstack dwarf` option. This will make async-profiler unwind native frames on its own instead of relying on the JVM. In this case, profiler should discover the first Java frame correctly, it will not accidentally hit a zombie method.
